### PR TITLE
[numbers] support JUnit 5 tests

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

**Comment by aherbert** [comment]
I think the recommended way is to use the JUnit [Bill of Materials](https://junit.org/junit5/docs/current/user-guide/#dependency-metadata-junit-bom) which contains all the versions which work together:

```
  <dependencyManagement>
      <dependency>
        <groupId>org.junit</groupId>
        <artifactId>junit-bom</artifactId>
        <!-- define this in properties -->
        <version>${numbers.junit.bom.version}</version>
        <scope>import</scope>
        <type>pom</type>
      </dependency>
    </dependencies>
  </dependencyManagement>

  <!-- Include what you require without version numbers. 
       They are all in the BOM. -->
  <dependencies>
    <dependency>
      <groupId>org.junit.jupiter</groupId>
      <artifactId>junit-jupiter</artifactId>
      <scope>test</scope>
    </dependency>
    <!-- For IDE support --> 
    <dependency>
      <groupId>org.junit.platform</groupId>
      <artifactId>junit-platform-launcher</artifactId>
      <scope>test</scope>
    </dependency>
    <!-- For JUnit 4 support --> 
    <dependency>
      <groupId>org.junit.platform</groupId>
      <artifactId>junit-platform-runner</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.junit.vintage</groupId>
      <artifactId>junit-vintage-engine</artifactId>
      <scope>test</scope>
    </dependency>
  </dependencies>

```

You can see what it bundles on Maven repository [org.junit:junit-bom:5.4.2](https://mvnrepository.com/artifact/org.junit/junit-bom/5.4.2).

I am not sure how it works with the legacy junit 4 dependency. I would try deleting that from the POM and just trying with the vintage engine dependency.

**Comment by grimreaper** [comment]
@aherbert what about this iteration? aherbert

(also, you're allowed to push to this branch if it'd help)

File: pom.xml [hunk 1]
```diff
@@ -171,6 +188,7 @@
     <allowscript.javadoc.qualifier></allowscript.javadoc.qualifier>
     <!-- Invalid flag for old javadoc versions (default to empty). -->
     <doclint.javadoc.qualifier></doclint.javadoc.qualifier>
+    <numbers.junit.bom.version>5.4.2</numbers.junit.bom.version>
```
Comment (aherbert) [inline pom.xml]: I would move this to the location of other versions, e.g. below numbers.mathjax.version

File: pom.xml [hunk 2]
```diff
@@ -96,14 +96,31 @@
         <type>test-jar</type>
         <scope>test</scope>
       </dependency>
+      <dependency>
+        <groupId>org.junit</groupId>
+        <artifactId>junit-bom</artifactId>
+        <!-- define this in properties -->
```
Comment (aherbert) [inline pom.xml]: The comment can be removed. It was laziness when I posted the example to not also define a properties section.

**Comment by aherbert** [comment]
Looks fine. Let's see if Travis can run all the tests.

File: pom.xml (hunk 2 above)
Comment (grimreaper) [reply pom.xml]: and it was further lazyness to copy and paste it when I updated :)

**Comment by grimreaper** [comment]
Updated. Travis failed previously on coveralls related errors but not on the core tests.

**Comment by grimreaper** [comment]
Pushed as fccc6fee3d69b42b1f201a0ff9458131ccfcc5f0. Hope I did that right.

**Comment by aherbert** [comment]
I've pulled master and the changes are there. Downstream PRs for the switch to JUnit 5 should all be OK.

**Comment by grimreaper** [comment]
actually see 6bcc6d9a16bea265243f68b298713d10abb09224

**Pull request closed without merging**